rfd/0029-account-lifecycle.md

```diff
@@ -0,0 +1,231 @@
+tokens are invalid.
+
+**Internal request for canceling the account**
+
+To request account cancellation, any account owner should:
+
+* Be logged into Teleport UI.
+* Press a "cancel my account" button in the UI.
+* UI should verify that they own a second factor.
+* Verify that they have write permission on the billing.
+
+```
+We have marked your account as inactive. In 10 days, we are going to delete the account and all it's data.
+To prevent this, log into Teleport and press "revert account cancellation".
+```
+
+The account enters inactive state:
+
+**Inactive account state**
+
+* Teleport should not disconnect any resources connected to the cloud account during the grace period.
+Otherwise it is possible for attacker overtaking an account owner to cause system wide outage.
+* Account owner will be able to login in the account, but will
+see message of the day displayed in UI and CLI:
+
+```
+Your account was marked as inactive at the request of `email`.
+In 10 days, we are going to delete the account and all it's data.
+```
+
+* Any account owner can login into the account and is greeted with a warning:
+
+```
+Your account was marked as inactive at the request of `email`.
+To prevent account from being deleted, go to `My Account` and activate `Revert account cancellation`.
+```
+
+All other owner users should receive the email with the message:
+
+```
+Your account was marked as inactive at the request of `email`.
+To prevent account from being deleted, login into teleport, go to `My Account` and activate `Revert account cancellation`.
+```
+
+This email notification should be sent every 3 days to all account owners until the account is deleted.
+
+Any account owner should be able to revert account cancellation.
+
+Teleport will ask logged in account owner for a second factor proof to revert account cancellation to prevent
+a possible account overtake by a malicious actor who wishes to keep the account for accessing
+the resources.
+
+Once system leaves inactive account state, all account owners
+should receive an email:
+
+```
+Your account was marked as active at the request of `email`.
+```
+
+The possible vectors of attack is for malicious actor to overtake someone's
+account with a request to delete all the data. The grace period and active privileged
+user should mitigate against that. Email sent to other account owners
+should alert other owners about account cancellation.
+
+If there are no other owners, other active users will notice the message of the day in the UI and CLI.
+
+Teleport cloud support should be able to delay account cancellation by 10 days if they see a zendesk
+issue from any zendesk account.
+
+**When all is lost*
+
+This flow is only relevant for Cloud.
+
+If any user or account owner used their and other shared email address, they won't be able to recover
+their account with us, but can request cancellation. There is no way for Teleport to associate users
+with social accounts and the account or company name.
```

Comment:
What kinds of email addresses are limited here (the phrasing is a bit confusing)?

Let's say I'm a hobbyist trying out Teleport Cloud.
I sign up with my private email address.
Why can't Teleport Cloud send an account reset link to that same address?

Reply:
If we send an account reset link, this will always downgrade all our two factor authentication to just one factor - email.

So the attacker would not bother hacking any of our two factors, they would just need to hack the email address. 
SMTP protocol is not very secure either - it does not guarantee encryption all the way. The attacker could perform a downgrade attack or MITM. [MTA-STS](https://tools.ietf.org/html/rfc8461) could improve things, but still :)

Reply:
```suggestion
If any user or account owner lost their password, second factor and recovery tokens, they won't be able to recover
their account with us, but can request cancellation. We cannot trust account recovery using email verification alone.
```